### PR TITLE
[bugfix] getDistinctTypeHierarchy didn't expand interface hierarchy.

### PR DESCRIPTION
**Pull request by Laimiux**
Reflection.getDistinctTypeHierarchy now handles the following use case.
```kotlin
interface First extends Second {}
interface Second extends Third {}
interface Third {}
```

### CODE REVIEW

**Comment by JakeWharton** [comment]
Can we test for this? Otherwise it'll likely regress. Maybe a module with that interface hierarchy and a static provider on `Third`.

**Comment by Laimiux** [comment]
Added a test, however `code gen` variant fails because it cannot find `DaggerNestedDependencyInterfaceTest` class.

File: integration-tests/src/test/java/com/example/NestedDependencyInterfaceTest.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
+package com.example;
+
+import dagger.Component;
+
+@Component(dependencies = NestedDependencyInterfaceTest.First.class)
+public interface NestedDependencyInterfaceTest {
```
Comment (TWiStErRob) [inline integration-tests/src/test/java/com/example/NestedDependencyInterfaceTest.java]: `DaggerNestedDependencyInterfaceTest` is generated by traditional dagger, which is only executed on production code:
https://github.com/JakeWharton/dagger-reflect/blob/b009ca8d3cdd6dac0a6c653ea4459e02996a540e/integration-tests/build.gradle#L8

Move this class to `src/main/java`.

**Comment by JakeWharton** [comment]
Rebased, moved the test subject to `src/main/`, removed `@IgnoreCodegen`, and merged. Thanks!

**Pull request closed without merging**